src/chip8-solns.md

```diff
@@ -29,29 +31,32 @@ You can add any other modules, for tests or anything else, anywhere you wish.
 The interpreter/CPU/virtual machine struct should look something like this:
 
 ```rust,noplayground
-pub struct VM {
+pub struct ChipState {
     memory: [u8; 4096],
-    pc: u16,
+    program_counter: u16,
     registers: [u8; 16],
     display: chip8_base::Display,
-    stack: Vec<u16>
+    stack_pointer: u8,
+    stack: [u16; 16],
+    // ... there will be more
 }
 ```
 
 Only a few of the fields you need are included here, you'll need to add a few more as you go, and you can represent them however you wish. The corresponding `new()` method should look like this:
 
 ```rust,noplayground
-impl VM {
-    pub fn new() -> Self {
-        Cpu {
+impl ChipState {
+    pub fn new(freq: u32) -> Self {
+        Self { 
             memory: [0; 4096],
-            pc: 0,
-            registers: [0; 16],
-            display: [[0; 64]; 32]
-            stack: Vec::new(),
+            registers: [0; 16], 
+            program_counter: 0x200,
+            display: [[0; 64]; 32],
+            stack_pointer: 0,
+            stack: [0; 16],
```

Comment:
The spec linked limits it to 16. A `Vec` is definitely the generally better choice, but I felt it was better to follow the spec exactly, since that's what people will be reading it from

http://devernay.free.fr/hacks/chip8/C8TECH10.HTM#2.2 last sentence of this section

Reply:
Oh, didn't clock that. that's cool then. Having the stack pointer as an actual reference though could be interesting., not sure how it would play with lifetimes.

Reply:
Nevermind on that 2nd point, that would be a self referential struct, which is hard (https://doc.rust-lang.org/nightly/std/pin/index.html#example-self-referential-struct)